dashboard/app/assets/javascripts/small_footer.js.jsx

```diff
@@ -0,0 +1,228 @@
+/* $ */
+
+// This code might better live in shared eventually. doing that would
+// require adding JSX transpiling to shared, and the ability to output multiple
+// bundles
+
+window.dashboard = window.dashboard || {};
+
+var SmallFooter;
+
+/**
+ * Gets our React component SmallFooter. We use a getter so that we don't depend
+ * on React being in the global namespace at load time.
+ */
+window.dashboard.getSmallFooterComponent = function (React) {
+  if (SmallFooter) {
+    return SmallFooter;
+  }
+
+  var MenuState = {
+    MINIMIZING: 'MINIMIZING',
+    MINIMIZED: 'MINIMIZED',
+    EXPANDED: 'EXPANDED',
+    COPYRIGHT: 'COPYRIGHT'
+  };
+
+  var EncodedParagraph = React.createClass({
+    render: function () {
+      return <p dangerouslySetInnerHTML={{
+          __html: decodeURIComponent(this.props.text)
+      }}/>
+    }
+  });
+
+  return SmallFooter = React.createClass({
+    propTypes: {
+      // We let dashboard generate our i18n dropdown and pass it along as an
+      // encode string of html
+      i18nDropdown: React.PropTypes.string,
+      copyrightInBase: React.PropTypes.bool.isRequired,
+      copyrightStrings: React.PropTypes.shape({
+        thank_you: React.PropTypes.string.isRequired,
+        help_from_html: React.PropTypes.string.isRequired,
+        art_from_html: React.PropTypes.string.isRequired,
+        powered_by_aws: React.PropTypes.string.isRequired,
+        trademark: React.PropTypes.string.isRequired
+      }),
+      baseCopyrightString: React.PropTypes.string,
+      baseMoreMenuString: React.PropTypes.string.isRequired,
+      baseStyle: React.PropTypes.object,
+      menuItems: React.PropTypes.arrayOf(
+        React.PropTypes.shape({
+          text: React.PropTypes.string.isRequired,
+          link: React.PropTypes.string.isRequired
+        })
+      ).isRequired,
+      className: React.PropTypes.string
+    },
+
+    getInitialState: function () {
+      return {
+        menuState: MenuState.MINIMIZED,
+        copyrightStyle: null,
+        moreMenuStyle: null
+      };
+    },
+
+    componentDidMount: function () {
+      var originalSetState = this.setState;
+      var base = React.findDOMNode(this.refs.base);
+      this.setState({
+        copyrightStyle: {
+          paddingBottom: base.offsetHeight
+        },
+        moreMenuStyle: {
+          bottom: base.offsetHeight,
+          width: base.offsetWidth,
+        }
+      });
+    },
+
+    minimizeOnClickAnywhere: function (event) {
+      // The first time we click anywhere, hide any open children
+      $(document.body).one('click', function (event) {
```

Comment:
diabolical test case to please check manually: 
(1) run your program
(2) open the footer
(3) hover over a block in the toolbox, or type the partial name of a command in text mode
(4) click "see examples"
(5) click somewhere else

EXPECTED: the footer menu goes away after the second click

I ask because the "see examples" click calls stop propagation. I realize this is a bit of a long shot, but it would suck if there were a circumstance where the footer stayed open forever.